# HG changeset patch
# User Patrick Bruenn <p.bruenn@beckhoff.com>
# Date 1511938675 -3600
#      Wed Nov 29 07:57:55 2017 +0100
# Branch stable-1.5
# Node ID 5ac7a113010964a82723795cfd231ba3a70ac438
# Parent  9149adda56ed7bc600a263f9a7639918eeb6d117
ccat: fix netdev for kernel >= 4.11

diff -r 9149adda56ed -r 5ac7a1130109 devices/ccat/netdev.c
--- a/devices/ccat/netdev.c	Fri Oct 27 12:33:30 2017 +0200
+++ b/devices/ccat/netdev.c	Wed Nov 29 07:57:55 2017 +0100
@@ -22,6 +22,7 @@
 #include <linux/kernel.h>
 #include <linux/module.h>
 #include <linux/netdevice.h>
+#include <linux/version.h>
 
 #ifdef CONFIG_PCI
 #include <asm/dma.h>
@@ -793,8 +794,12 @@
 	return HRTIMER_RESTART;
 }
 
+#if (LINUX_VERSION_CODE < KERNEL_VERSION(4,11,0))
 static struct rtnl_link_stats64 *ccat_eth_get_stats64(struct net_device *dev, struct rtnl_link_stats64
 						      *storage)
+#else
+static void ccat_eth_get_stats64(struct net_device *dev, struct rtnl_link_stats64 *storage)
+#endif
 {
 	struct ccat_eth_priv *const priv = netdev_priv(dev);
 	struct ccat_mac_register mac;
@@ -829,7 +834,9 @@
 	/* for cslip etc */
 	//TODO __u64    rx_compressed;
 	//TODO __u64    tx_compressed;
+#if (LINUX_VERSION_CODE < KERNEL_VERSION(4,11,0))
 	return storage;
+#endif
 }
 
 static int ccat_eth_open(struct net_device *dev)
